include/envoy/config/typed_metadata.h

```diff
@@ -65,6 +65,16 @@ class TypedMetadataFactory : public UntypedFactory {
   virtual std::unique_ptr<const TypedMetadata::Object>
   parse(const ProtobufWkt::Struct& data) const PURE;
 
+  // Subclass has to override and implement this function so that Envoy will be
+  // able to parse and use `typed_filter_metadata` field. Otherwise,
+  // `typed_filter_metadata` will be ignored.
+  // This is not pure virtual because we do not want to break any existing
```

Comment:
I think this class is used in internal implementation only. It should be ok to add one more `PURE` method and modify 3 derived classes to include the new method.

Reply:
Yes, make sense.